backend/game/cards/keep_card.py

```diff
@@ -7,11 +7,11 @@ def __init__(self, name, cost, effect):
         super().__init__(name, cost, effect)
 
     def immediate_effect(self, player_that_bought_the_card, other_players):
-        player_that_bought_the_card.add_card(self)
```

Comment:
@gpfaia have a future refactor planned to remove the immediate_effect and discard_effect because it looked like they were originally only used to draw/discard the cards. For the meantime we were planning to leave this class alone and get the old stale pull requests updated and pushed through so we don't have to repeat the refactor on all branches.